api/osf_api.py

```diff
@@ -197,17 +183,46 @@ def get_existing_file(session, node_id=settings.PREFERRED_NODE):
         return upload_fake_file(session, node)
 
 
-def upload_fake_file(session, node=None, name='osf selenium test file for testing because its fake.txt', upload_url=None, provider='osfstorage'):
+def upload_single_quickfile(session):
+    """Upload a new quickfile. Delete existing quickfiles first.
+    Note: Currently using v2.0 of the API. Certain lines will need to be changed on update.
+    TODO: Make this more general.
+    """
+    user = current_user(session)
+    quickfiles_url = user.relationships.quickfiles['links']['related']['href']
+    delete_all_quickfiles(session, quickfiles_url)
+
+    upload_url = user.relationships.quickfiles['links']['upload']['href']
+    return upload_fake_file(session, upload_url=upload_url, quickfile=True)
+
+
+def delete_all_quickfiles(session, quickfiles_url):
+    """ Delete all quickfiles. Just pass in the quickfiles url for the currently logged in user.
+    """
+    for quickfile in session.get(quickfiles_url)['data']:
+        delete_url = quickfile['links']['delete']
+        delete_file(session, delete_url)
+
+
+def upload_fake_file(session, node=None, name='osf selenium test file for testing because its fake.txt', upload_url=None, provider='osfstorage', quickfile=False):
     """Upload an almost empty file to the given node. Return the file's name.
 
     Note: The default file has a very long name because it makes it easier to click a link to it.
+    Quickfiles Note: "/?create_guid=1" must be added via api BEFORE viewing on the front end. See ENG-1351 for more info
     """
     if not upload_url:
         if not node:
             raise TypeError('Node must not be none when upload URL is not set.')
         upload_url = '{}/v1/resources/{}/providers/{}/'.format(settings.FILE_DOMAIN, node.id, provider)
 
     metadata = session.put(url=upload_url, query_parameters={'kind': 'file', 'name': name}, raw_body={})
+
+    if quickfile:
+        user = current_user(session)
+        quick_files_url = user.relationships.quickfiles['links']['related']['href']
+        info_link = session.get(quick_files_url)['data'][0]['links']['info']
+        session.get(info_link + '?create_guid=1')
```

Comment:
* Can we apply the `create_guid` param to the upload `session.put()` so we don't have to do the extra `get` request?
* Appending query params directly may not work all the time: `info_link + '?create_guid=1'`. e.g. If the base URL already have query parameters then we need `&` instead of `?`; or the query param sometimes needs to be URL encoded. Use something like `query_parameters={'create_guid': 1, }` instead so we don't need to worry about the full URL building complication.

Reply:
> Can we apply the create_guid param to the upload session.put() so we don't have to do the extra get request?

A quick peruse of the code suggests "No".  It [looks like](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/files/views.py#L105) the `create_guid` param is tied to the `GET` request.

Reply:
So maybe a comment here so we don't have the same question later?

Reply:
Thanks for the input guys. I've added this comment:

> create_guid param is tied to the GET request so we can't use `query_parameters={'create_guid': 1}` here